Assets/PatchKit Patcher/Scripts/AppData/Local/LocalMetaData.cs

```diff
@@ -119,12 +120,24 @@ public void RegisterEntry([NotNull] string entryName, int versionId, long entryS
             }
         }
 
-        private bool ShouldSaveEntry(long entrySize)
+        private bool ShouldSaveEntry(long? entrySize)
         {
-            if (_unsavedEntriesSize > UnsavedEntriesSizeLimit)
+            if (entrySize.HasValue)
             {
-                return true;
+                _unsavedEntriesSize += entrySize.Value;
+                if (_unsavedEntriesSize > UnsavedEntriesSizeLimit)
+                {
+                    return true;
+                }
             }
+            else
+            {
+                if (++_unsavedEntriesCount > UnsavedEntriesCountLimit)
+                {
+                    return true;
+                }
```

Comment:
Should be checked regardless `entrySize` being `null`.

Reply:
done